discovery/uyuni/uyuni.go

```diff
@@ -206,135 +192,107 @@ func NewDiscovery(conf *SDConfig, logger log.Logger) *Discovery {
-						for _, g := range candidateGroups {
-							// get list of group formulas
-							// TODO: Put the resulting data on a map so that we do not have to repeat the call below for every system
-							if g.Subscribed == 1 {
-								groupFormulas, err := getGroupFormulaData(rpcclient, token, g.ID, "prometheus-exporters")
-								if err != nil {
-									level.Error(d.logger).Log("msg", "getGroupFormulaData failed", "groupId", client.ID, "err", err)
-									return
-								}
-								formulas = getCombinedFormula(formulas, groupFormulas)
-								// replace spaces with dashes on all group names
-								groups = append(groups, strings.ToLower(strings.ReplaceAll(g.SystemGroupName, " ", "-")))
-							}
-						}
-
-						// Get system formula list
-						systemFormulas, err := getSystemFormulaData(rpcclient, token, client.ID, "prometheus-exporters")
-						if err != nil {
-							level.Error(d.logger).Log("msg", "getSystemFormulaData failed", "clientId", client.ID, "err", err)
-							return
-						}
-						formulas = getCombinedFormula(formulas, systemFormulas)
-
-						// Iterate list of formulas and check for enabled exporters
-						for k, v := range formulas {
-							if v.Enabled {
-								port, err := extractPortFromFormulaData(v.Args)
-								if err != nil {
-									level.Error(d.logger).Log("msg", "Invalid exporter port", "clientId", client.ID, "err", err)
-									return
-								}
-								targets := model.LabelSet{}
-								addr := fmt.Sprintf("%s:%s", netInfo.IP, port)
-								targets[model.AddressLabel] = model.LabelValue(addr)
-								targets["exporter"] = model.LabelValue(k)
-								targets["hostname"] = model.LabelValue(details.Hostname)
-								targets["groups"] = model.LabelValue(strings.Join(groups, ","))
-								for _, g := range groups {
-									gname := fmt.Sprintf("grp_%s", g)
-									targets[model.LabelName(gname)] = model.LabelValue("active")
-								}
-								tg.Targets = append(tg.Targets, targets)
-							}
-						}
-					}
-				}
-				// Log debug information
-				if netInfo.IP != "" {
-					level.Info(d.logger).Log("msg", "Found monitored system", "Host", details.Hostname,
-						"Entitlements", fmt.Sprintf("%+v", details.Entitlements),
-						"Network", fmt.Sprintf("%+v", netInfo), "Groups",
-						fmt.Sprintf("%+v", groups), "Formulas", fmt.Sprintf("%+v", formulas))
-				}
-				rpcclient.Close()
-			}(cl)
+	targets := make([]model.LabelSet, 0)
+	if len(systemGroupIDsBySystemID) > 0 {
+		targetsForSystems, err := d.getTargetsForSystems(rpcClient, token, systemGroupIDsBySystemID)
+		if err != nil {
+			return nil, err
 		}
-		wg.Wait()
+		targets = append(targets, targetsForSystems...)
 		level.Info(d.logger).Log("msg", "Total discovery time", "time", time.Since(startTime))
+	} else {
+		fmt.Printf("\tFound 0 systems.\n")
 	}
-	logout(rpc, token)
-	rpc.Close()
-	return []*targetgroup.Group{tg}, nil
+
+	level.Info(d.logger).Log("msg", "Targets", "targets", fmt.Sprintf("%+v", targets))
```

Comment:
This list can also grow very big. Maybe switch level to debug and add an extra entry that logs in 'info' level the total number of targets?